src/sql/workbench/contrib/notebook/browser/calloutDialog.ts

```diff
@@ -0,0 +1,309 @@
+const IMAGE = 'IMAGE_PREVIEW';
+
+export interface ICalloutDialogOptions {
+	insertTtitle?: string,
+	calloutStyle?: CalloutStyle,
+	insertMarkup?: string,
+	imagePath?: string,
+	embedImage?: boolean
+}
+
+export class CalloutDialog extends Modal {
+	private _calloutStyle: CalloutStyle;
+	private _selectionComplete: Deferred<ICalloutDialogOptions>;
+	private _insertButton: Button;
+	private _cancelButton: Button;
+	// Link
+	private _linkTextLabel: HTMLElement;
+	private _linkTextInputBox: InputBox;
+	private _linkAddressLabel: HTMLElement;
+	private _linkAddressInputBox: InputBox;
+	// Image
+	private _imageLocationLabel: HTMLElement;
+	private _imageLocalRadioButton: RadioButton;
+	private _imageRemoteRadioButton: RadioButton;
+	private _imageUrlLabel: HTMLElement;
+	private _imageUrlInputBox: InputBox;
+	private _imageBrowseButton: HTMLAnchorElement;
+	private _imageEmbedLabel: HTMLElement;
+	private _imageEmbedCheckbox: Checkbox;
+
+	private readonly insertButtonText = localize('callout.insertButton', "Insert");
+	private readonly cancelButtonText = localize('callout.cancelButton', "Cancel");
+	// Link
+	private readonly linkTextLabel = localize('callout.linkTextLabel', "Text to display");
+	private readonly linkTextPlaceholder = localize('callout.linkTextPlaceholder', "Text to display");
+	private readonly linkAddressLabel = localize('callout.linkAddressLabel', "Address");
+	private readonly linkAddressPlaceholder = localize('callout.linkAddressPlaceholder', "Link to an existing file or web page");
+	// Image
+	private readonly locationLabel = localize('callout.locationLabel', "Image location");
+	private readonly localImageLabel = localize('callout.localImageLabel', "This computer");
+	private readonly remoteImageLabel = localize('callout.remoteImageLabel', "Online");
+	private readonly pathInputLabel = localize('callout.pathInputLabel', "Image URL");
+	private readonly pathPlaceholder = localize('callout.pathPlaceholder', "Enter image path");
+	private readonly urlPlaceholder = localize('callout.urlPlaceholder', "Enter image URL");
+	private readonly browseAltText = localize('callout.browseAltText', "Browse");
+	private readonly embedImageLabel = localize('callout.embedImageLabel', "Attach image to notebook");
+
+	constructor(
+		calloutInstance: CalloutStyle,
+		title: string,
+		posX: number,
+		posY: number,
+		@IFileDialogService private readonly _fileDialogService: IFileDialogService,
+		@IThemeService themeService: IThemeService,
+		@ILayoutService layoutService: ILayoutService,
+		@IAdsTelemetryService telemetryService: IAdsTelemetryService,
+		@IContextKeyService contextKeyService: IContextKeyService,
+		@IContextViewService private contextViewService: IContextViewService,
+		@IClipboardService clipboardService: IClipboardService,
+		@ILogService logService: ILogService,
+		@ITextResourcePropertiesService textResourcePropertiesService: ITextResourcePropertiesService
+	) {
+		super(
+			title,
+			TelemetryKeys.SelectImage,
```

Comment:
not sure how is this dialog related to SelectImage

Reply:
calloutDialog is called by src/sql/workbench/contrib/notebook/browser/markdownToolbarActions.ts -- the individual toolbar actions, including Insert Image and Insert Link are handled here.

Reply:
I would think that a better name would be a general `NotebookCallout` or something if it's doing more than just selecting an image though. 

Or at least changing the name to be different for the different types of callout being used here

Reply:
Changed to: NotebookCallout